cowprotocol/cowswap
# fix: don't reset network cache in JsonRpcProvider

# Summary

`useBlockNumber()` subscribes on block and fetch them on every block, besides of that it also fetches `eth_chainId` what is excessive. To avoid that I patched `json-rpc-provider.js` to prevent network cache resetting. For some reason in the lib they cache it only for the current event loop.

Before:
<img width="1193" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/d095471f-f56a-43c7-8507-4f782f9f0a99">

After:
<img width="1191" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/0c855fca-fa17-421e-9c77-dfd73f3d4fb8">


  # To Test

1. Open dapp
2. Open dev tools with network tab
- [ ] AR: there are batches by two requests with `useBlockNumber` initiator
- [ ] ER: there are single requests with `useBlockNumber` initiator
3. Network switching should work (with not connected wallet)
4. Network switching should work (with connected wallet)
